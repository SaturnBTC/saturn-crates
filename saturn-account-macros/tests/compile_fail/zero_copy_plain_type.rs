use saturn_account_macros::Accounts;

#[derive(bytemuck::Pod, bytemuck::Zeroable, Copy, Clone)]
#[repr(C)]
pub struct Data {
    pub value: u64,
}

#[derive(Accounts)]
struct ZeroCopyPlain<'info> {
    #[account(zero_copy, of = Data)]
    acc: arch_program::account::AccountInfo<'info>,
}

fn main() {} 