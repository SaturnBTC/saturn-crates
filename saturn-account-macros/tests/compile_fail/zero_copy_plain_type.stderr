error[E0599]: the function or associated item `new` exists for struct `AccountLoader<'_, AccountInfo<'info>>`, but its trait bounds were not satisfied
  --> tests/compile_fail/zero_copy_plain_type.rs:10:10
   |
10 | #[derive(Accounts)]
   |          ^^^^^^^^ function or associated item cannot be called on `AccountLoader<'_, AccountInfo<'info>>` due to unsatisfied trait bounds
   |
  ::: $WORKSPACE/deps/program/src/account.rs
   |
   | pub struct AccountInfo<'a> {
   | -------------------------- doesn't satisfy `AccountInfo<'info>: Pod` or `AccountInfo<'info>: Zeroable`
   |
   = note: the following trait bounds were not satisfied:
           `AccountInfo<'info>: Pod`
           `AccountInfo<'info>: Zeroable`
   = note: this error originates in the derive macro `Accounts` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0277]: the trait bound `AccountInfo<'info>: Pod` is not satisfied
  --> tests/compile_fail/zero_copy_plain_type.rs:10:10
   |
10 | #[derive(Accounts)]
   |          ^^^^^^^^ the trait `Pod` is not implemented for `AccountInfo<'info>`
   |
   = help: the following other types implement trait `Pod`:
             ()
             Data
             ManuallyDrop<T>
             PhantomData<T>
             PhantomPinned
             RuneAmount
             RuneId
             Wrapping<T>
           and $N others
note: required by a bound in `AccountLoader`
  --> $WORKSPACE/saturn-account-parser/src/codec/zero_copy.rs
   |
   | pub struct AccountLoader<'a, T>
   |            ------------- required by a bound in this struct
   | where
   |     T: Pod + Zeroable + 'static,
   |        ^^^ required by this bound in `AccountLoader`
   = note: this error originates in the derive macro `Accounts` (in Nightly builds, run with -Z macro-backtrace for more info)
